### PR TITLE
Addresses Static Web App deployment issues

### PR DESCRIPTION
**Pull request by wbreza**
Address a few different Static Web App deployment issues

- [x] Deprecates usage of `swa login`.  Will now manually query for deployment token and pass via `--deployment-token` param of `swa deploy`
- [x] Defaults to `default` environment name regardless of azd environment name till Azure/azure-dev-pr#1152 is resolved
- [x] Adds deployment validation check to ensure environment is in "Ready" state 
- [x] Now works correctly in Linux based hosts

### CODE REVIEW

File: cli/azd/test/samples/funcapp/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
-# Do not include azure-functions-worker as it may conflict with the Azure Functions platform
-
+# Do not include azure-functions-worker as it may conflict with the Azure Functions platform
```
Comment (wbreza) [inline cli/azd/test/samples/funcapp/requirements.txt:1]: This change appeared and won't go away...

**Review by vhvb1989 — APPROVED** [review]
add changelog entry

**Comment by vhvb1989** [comment]
Tested on Linux VM.
Fails if using node 14 LTS

Works as expected with node v16 (tested 16.0.0 and 16.16.0)

File: cli/azd/test/samples/funcapp/requirements.txt (hunk 1 above)
Comment (ellismg) [reply cli/azd/test/samples/funcapp/requirements.txt:1]: This is a whitespace only change - I suspect it's line ending stuff.

**Review by ellismg — APPROVED** [review]
LGTM

**Pull request merged**